modules/ding_frontend/ding_frontend.admin.inc

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/**
+ * @file
+ * Module's admin features.
+ */
+
+/**
+ * Administration form.
+ */
+function ding_frontend_settings($form, &$form_state) {
+  $types = array(
+    'ding_news' => t('News'),
+    'ding_event' => t('Event'),
+  );
+  $form['ding_frontend_fe_promote_types'] = array(
+    '#title' => t('Promote types to Frontpage'),
+    '#type' => 'checkboxes',
+    '#options' => $types,
+    '#default_value' => variable_get('ding_frontend_fe_promote_types', array('ding_event' => 'ding_event')),
```

Comment:
I re-reviewed the code and that is my only remaining comment.

In the current state the form does not seem to work. Changing the configuration does not change the default value. This is caused by never setting the variable `ding_frontend_fe_promote_types`. I do not think it is needed either.

As mentioned previously the default value in this form element should be retrieved from the value of the existing `node_options` variables. To avoid overriding configuration on deployment we should remove  the `node_options` variables from features and set their default values on install instead.